deeplake/util/keys.py

```diff
@@ -183,15 +184,36 @@ def get_sequence_encoder_key(key: str, commit_id: str) -> str:
 
 
 def dataset_exists(storage) -> bool:
+    """
+    Returns true if a dataset exists at the given location.
+    NOTE: This does not verify if it is a VALID dataset, only that it exists and is likely a deeplake directory.
+    To verify the content, use :func:`dataset_valid`
+    """
     try:
-        storage[get_dataset_meta_key(FIRST_COMMIT_ID)]
-        return True
+        return (
+                get_dataset_meta_key(FIRST_COMMIT_ID) in storage
+                or get_version_control_info_key() in storage
+        )
     except S3GetAccessError as err:
         raise AuthorizationException("The dataset storage cannot be accessed") from err
-    except (KeyError, S3GetError) as err:
+    except S3GetError:
         return False
 
 
+def dataset_validate(storage) -> None:
+    """
+    Checks the structure of the dataset and throws a descriptive DatasetCorruptError for any problems.
+    """
+    try:
+        for file in [get_dataset_meta_key(FIRST_COMMIT_ID), get_version_control_info_key()]:
+            if file not in storage:
+                raise DatasetCorruptError(f"Invalid dataset: missing file {file}")
+    except S3GetAccessError as err:
+        raise AuthorizationException("The dataset storage cannot be accessed") from err
+    except S3GetError:
```

Comment:
Same question here, does having other cloud datasets (GCS/Azure) impact the following error execution?

Reply:
is actually calling S3 related error handling good outside of storage provider abstraction?

Reply:
I wasn't a fan of the S3 exceptions showing at this level of the code either, but was sticking with the existing pattern for now. Ideally they'd all have standard exceptions thrown from a common interface, but trying to keep this fix straightforward for now.

Reply:
got it.